tools/ci/crate-range.py

```diff
@@ -179,12 +178,18 @@ def main():
 
     subparser_generate_matrix = subparsers.add_parser("generate-matrix", help="Generate a test matrix")
     subparser_generate_matrix.add_argument("-b", type=int, dest="batches", required=True, help="Number of batches")
+    subparser_generate_matrix.add_argument("-f", required=True, type=str, dest="folder",
+                                           help="Name of the folder containing the crates you want to generate a "
+                                                "matrix for")
     subparser_generate_matrix.add_argument("rust_versions", type=str, nargs=argparse.ONE_OR_MORE)
 
     subparser_run = subparsers.add_parser("run", help="Run command on crate range")
     subparser_run.add_argument("-b", required=True, type=int, dest="batches", help="Number of batches")
     subparser_run.add_argument("-s", required=True, type=int, dest="start_inclusive", help="Range start inclusive")
     subparser_run.add_argument("-e", required=True, type=int, dest="end_exclusive", help="Range end exclusive")
+    subparser_run.add_argument("-f", required=True, type=str, dest="folder", help="Name of the folder containing the "
+                                                                                  "crates you want to generate a "
+                                                                                  "matrix for") 
```

Comment:
is the arg intentionally in the list twice?

Reply:
It's for a different subcommand. You have to specify the same arguments to both `generate-matrix` and `run` because both use those arguments to determine which directories to include.